src/RA_Defs.h

```diff
@@ -79,39 +79,42 @@ class RARect : public RECT
 class ResizeContent
 {
 public:
-    enum AlignType
+    enum class AlignType
     {
-        NO_ALIGN,
-        ALIGN_RIGHT,
-        ALIGN_BOTTOM,
-        ALIGN_BOTTOM_RIGHT
+        Right,
+        Bottom,
+        BottomRight
     };
 
 public:
-    HWND hwnd;
-    POINT pLT;
-    POINT pRB;
-    AlignType nAlignType;
-    int nDistanceX;
-    int nDistanceY;
-    bool bResize;
-
-    ResizeContent(HWND parentHwnd, HWND contentHwnd, AlignType newAlignType, bool isResize)
+    HWND hwnd{};
+    POINT pLT{};
+    POINT pRB{};
+    AlignType nAlignType{};
+    int nDistanceX{};
+    int nDistanceY{};
+    bool bResize{};
+
+    explicit ResizeContent(_In_ HWND contentHwnd, _In_ AlignType newAlignType, _In_ bool isResize) noexcept :
+        hwnd{ contentHwnd },
+        nAlignType{ newAlignType },
+        bResize{ isResize }
     {
-        hwnd = contentHwnd;
-        nAlignType = newAlignType;
-        bResize = isResize;
-
         RARect rect;
-        GetWindowRect(hwnd, &rect);
+        auto check = ::GetWindowRect(hwnd, &rect);
+        assert(check != 0);
 
-        pLT.x = rect.left;	pLT.y = rect.top;
-        pRB.x = rect.right; pRB.y = rect.bottom;
+        pLT ={ rect.left, rect.top };
+        pRB ={ rect.right, rect.bottom };
 
-        ScreenToClient(parentHwnd, &pLT);
-        ScreenToClient(parentHwnd, &pRB);
+        HWND__* const restrict parentHwnd = ::GetParent(contentHwnd);
```

Comment:
The restrict qualifier asserts disjoint storage (no-alias). Meaning no other pointer points the same address in the same scope. It optimizes the generated assembly code. The purpose is to get it to have the same performance as FORTRAN.

The` HWND` parameter above could not be restrict qualified because it gets aliased by `hwnd `(same address). If this were C, it would cause undefined behavior if it were restrict qualified but MSVC just ignores it if it was wrong.

Reply:
Just to note the C++ versions of GCC and Clang use `__restrict__` instead of `__restrict`.